src/main/java/com/uber/cadence/internal/worker/ActivityTaskHandler.java

```diff
@@ -35,9 +35,30 @@
   final class Result {
 
     private final RespondActivityTaskCompletedRequest taskCompleted;
-    private final RespondActivityTaskFailedRequest taskFailed;
+    private final TaskFailedResult taskFailed;
     private final RespondActivityTaskCanceledRequest taskCancelled;
     private final RetryOptions requestRetryOptions;
+    private int attempt;
+    private Duration backoff;
+
+    public static class TaskFailedResult {
```

Comment:
Why public?

Reply:
POJOActivityTaskHandler will need to use this.